MobiFlight/MobiFlightFirmwareUpdater.cs

```diff
@@ -32,20 +33,57 @@ public static bool IsValidFirmwareFilepath(string filepath)
             return File.Exists(filepath);
         }
 
+        // This method would, in theory, return the bus number and address for the supplied module.
+        // Unforutnately I can't find any way to do that.
+
+        public static Tuple<int, int> GetBusNumberAndAddress(MobiFlightModule module)

```

Comment:
This is where to start reading.

Reply:
Cool work, but honestly and no offense: I don't like the strong dependency to the flash tool in the configuration.
As I described on Discord already, we should find a more flexible way to allow for adding additional boards without modifying the MobiFlight code every time. This would allow for something like a community board repository where you can simply drop the boards definition together with a updater file into the MobiFlight installation. The benefit of this would be that the community can come up with more exotic and (experimental) boards that implement the MobiFlight API without having to "officially" support them with the core project.

When we started to discuss custom boards, then you proposed a flag to say: i am not updateable. don't worry about me, I can update my firmware myself. And I find that idea still cool! Why do we have to put in these specific implementations. tomorrow there might be another board.

it is super cool that you were able to add it into mobiflight and maybe we could leave it now since we have done it, but i am sure there will be another board.

With the avr dude it is quite simple, we can put all config required for starting the script into the boards definition file. I understand the lookup is different for the pico, but MobiFlight at the moment needs a serial port. Can we not pass the serial port to a little wrapper exe that then does the lookup and starts the pico flasher correctly?

Reply:
It doesn't matter where the code lives because I can't find any way to take the serial port and do a lookup of the necessary values. Whether the code lives inside Mobiflight or as a separate tool is irrelevant if there's no way to do it in the first place ;)

After days of digging into this I'm convinced it isn't possible. The best we could do is provide a pico firmware file and ship a board.json for it with MobiFlight. People will have to manually flash.

Reply:
i think the manual flashing is totally acceptable.

Reply:
and for clarification: this now works or still doesn't? because in the initial comments it sounds as if had figured it out how to do it through MF

Reply:
> this now works or still doesn't?

Calling PicoTool works, but there is no way to get the address and bus to pass to PicoTool so it knows what device to flash.

Reply:
so it doesn't work with MF